ui-core/src/surveyjs/address-validation-modal-question.tsx

```diff
@@ -0,0 +1,180 @@
+  addressValidationResult: AddressValidationResult
+}
+
+const AddressValidationType = 'addressvalidation'
+export class QuestionAddressValidationModel extends Question {
+  getType() {
+    return AddressValidationType
+  }
+
+  get street1() {
+    return this.getPropertyValue('street1')
+  }
+
+  set street1(val: string) {
+    this.setPropertyValue('street1', val)
+  }
+
+  get street2() {
+    return this.getPropertyValue('street2')
+  }
+
+  set street2(val: string) {
+    this.setPropertyValue('street2', val)
+  }
+
+  get city() {
+    return this.getPropertyValue('city')
+  }
+
+  set city(val: string) {
+    this.setPropertyValue('city', val)
+  }
+
+  // 'state' is already used on questions :(
+  get stateProvince() {
+    return this.getPropertyValue('state')
+  }
+
+  set stateProvince(val: string) {
+    this.setPropertyValue('state', val)
+  }
+
+  get country() {
+    return this.getPropertyValue('country')
+  }
+
+  set country(val: string) {
+    this.setPropertyValue('country', val)
+  }
+
+  get postalCode() {
+    return this.getPropertyValue('postalCode')
+  }
+
+  set postalCode(val: string) {
+    this.setPropertyValue('postalCode', val)
+  }
+}
+
+ElementFactory.Instance.registerElement(AddressValidationType, name => {
+  return new QuestionAddressValidationModel(name)
+})
+
+Serializer.addClass(
+  AddressValidationType,
+  [{
+    name: 'street1',
+    category: 'general'
+  }, {
+    name: 'street2',
+    category: 'general'
+  }, {
+    name: 'city',
+    category: 'general'
+  }, {
+    name: 'stateProvince',
+    category: 'general'
+  }, {
+    name: 'postalCode',
+    category: 'general'
+  }, {
+    name: 'country',
+    category: 'general'
+  }],
+  () => new QuestionAddressValidationModel(''),
+  'question'
+)
+
+
+export class SurveyQuestionAddressValidation extends SurveyQuestionElementBase {
+  get question() {
+    return this.questionBase
+  }
+
+  get value(): AddressValidationQuestionValue {
+    return this.question.value
+  }
+
+  accept(addr: MailingAddress) {
+    this.tryUpdateOtherValue(this.question.street1, addr.street1)
+    this.tryUpdateOtherValue(this.question.street2, addr.street1)
+    this.tryUpdateOtherValue(this.question.city, addr.city)
+    this.tryUpdateOtherValue(this.question.stateProvince, addr.state)
+    this.tryUpdateOtherValue(this.question.postalCode, addr.postalCode)
+    this.tryUpdateOtherValue(this.question.country, addr.country)
+  }
+
+  tryUpdateOtherValue(questionName: string, value: string) {
+    if (isEmpty(questionName)) {
+      return
+    }
+
+    const question = this.questionBase.survey.getQuestionByName(questionName)
+
+    if (question) {
+      question.value = value
+    }
+  }
+
+  renderElement() {
+    return (
+      <>
+        {(this.value.addressValidationResult?.suggestedAddress
+            && !this.value.canceledSuggestedAddress
+            && !this.value.acceptedSuggestedAddress) &&
+            <SuggestBetterAddressModal
```

Comment:
Note: This modal currently does not have any theming in the participant UI. It also is not based on any mocks from Erin, so it is definitely not ready for production.

Adding theming requires making it a `ThemedModal` using the participant-ui component, which, since this is in core, is a non-trivial task. I plan on revisiting this in another PR.